core/src/main/java/org/apache/brooklyn/entity/group/SshCommandMembershipTrackingPolicy.java

```
@@ -0,0 +1,105 @@
+
+    public static final MapConfigKey<Object> SHELL_ENVIRONMENT = BrooklynConfigKeys.SHELL_ENVIRONMENT;
+
+    public static final ConfigKey<String> UPDATE_COMMAND = ConfigKeys.newStringConfigKey("update.command", "Command to run on membership change events");
+
+    /**
+     * Called when a member is updated or group membership changes.
+     */
+    @Override
+    protected void onEntityEvent(EventType type, Entity entity) {
+        LOG.trace("Event {} received for {} in {}", new Object[] { type, entity, getGroup() });
+        String command = config().get(UPDATE_COMMAND);
+        if (Strings.isNonBlank(command)) {
+            execute(command);
+        }
+    }
+
+    public void execute(String command) {
+        Maybe<SshMachineLocation> machine = Machines.findUniqueMachineLocation(entity.getLocations(), SshMachineLocation.class);
+        if (machine.isAbsentOrNull()) {
+            throw new IllegalStateException("No machine available to execute command");
+        }
+        LOG.info("Executing command on {}: {}", machine.get(), command);
+        String executionDir = config().get(EXECUTION_DIR);
+        String sshCommand = SshCommandSensor.makeCommandExecutingInDirectory(command, executionDir, entity);
+
+        // Set things from the entities defined shell environment, overriding with our config
+        Map<String, Object> env = MutableMap.of();
+        env.putAll(MutableMap.copyOf(entity.config().get(BrooklynConfigKeys.SHELL_ENVIRONMENT)));
+        env.putAll(MutableMap.copyOf(config().get(BrooklynConfigKeys.SHELL_ENVIRONMENT)));
+
+        // Try to resolve the configuration in the env Map
+        try {
+            env = (Map<String, Object>) Tasks.resolveDeepValue(env, Object.class, ((EntityInternal) entity).getExecutionContext());
+        } catch (InterruptedException | ExecutionException e) {
+            Exceptions.propagateIfFatal(e);
```

Never swallow exceptions. We should log the exception if we're not propagating it, yes?